src/trix/models/composition.coffee

```diff
@@ -76,17 +76,25 @@ class Trix.Composition extends Trix.BasicObject
     document = @document
     {index, offset} = document.locationFromPosition(position)
     block = document.getBlockAtIndex(index)
+    previousCharacter = block.getPreviousCharacter(offset)
+    nextCharacter = block.getNextCharacter(offset)
 
     if block.getBlockBreakPosition() is offset
-      document = document.removeTextAtRange(range)
-      range = [position, position]
+      if previousCharacter is "\n"
+        document = document.removeTextAtRange([position - 1, position])
+      else if offset - 1 isnt 0
+        position += 1
     else
-      if block.text.getStringAtRange([offset, offset + 1]) is "\n"
+      if nextCharacter is "\n"
         range = [position - 1, position + 1]
       else if offset - 1 isnt 0
         position += 1
 
     newDocument = new Trix.Document [block.removeLastAttribute().copyWithoutText()]
+    if block.getConfig("breakOnReturn")
+      document = document.removeTextAtRange([position - 1, position])
+      range = [position - 1, position]
```

Comment:
The document is altered similarly above in the `if block.getBlockBreakPosition() is offset` condition. Will this end up removing text twice?